### ISSUE TITLE
Hooks do not fire reliably with "+" option

### ISSUE THREAD

**Issue by briankung** [zapier/django-rest-hooks#21]
I noticed that hooks with the "+" option were not being triggered for all users as I'd expected from the documentation - at least, not reliably.

From the documentation:
> If you want a Hook to be triggered for all users, add '+' to built-in Hooks or pass user_override=False for custom_hook events

However, in `utils.py`, `user_override` looks like it is set to `True` if there is a plus sign: https://github.com/zapier/django-rest-hooks/blob/master/rest_hooks/utils.py#L82

```py
            if model == maybe_model and action == maybe_action[0]:
                event_name = maybe_event_name
                if len(maybe_action) == 2:
                    user_override = True

    if event_name:
        find_and_fire_hook(event_name, instance, user_override=user_override)
```
In `find_and_fire_hook`, meanwhile, `user_override=True` is not accounted for:
```py
    if user_override is not False:
        if user_override:
            filters['user'] = user_override
        elif hasattr(instance, 'user'):
            filters['user'] = instance.user
        elif isinstance(instance, User):
            filters['user'] = instance
        else:
            raise Exception(
                '{} has no `user` property. REST Hooks needs this.'.format(repr(instance))
            )

       # ...comments...

    hooks = Hook.objects.filter(**filters)
    for hook in hooks:
        hook.deliver_hook(instance)
```
Setting `filters['user'] = user_override` results in `filters['user'] = True`. This means that `Hook.objects.filter(**filters)` will filter for `user=True`. I was very confused what that meant - I thought it reasonable to assume that it would _might_ be equivalent to `user__isnull=False`, but the actual result is stranger. I tested it with a few of our own models.

It turns out that if you ask for `SoAndSoModel.objects.filter(related_object=True)`, Django casts `related_object=True` to `related_object_id=1` and `related_object=False` to `related_object_id=0` in the query.

```py
pprint(connection.queries)
# For SoAndSoModel.objects.filter(related_object=True)
[{'sql': 'SELECT "thing_SoAndSoModel"."id", '
         # ...
         'WHERE "thing_SoAndSoModel"."related_object_id" = 1 LIMIT 21',
  'time': '0.004'},

# For SoAndSoModel.objects.filter(related_object=False)
 {'sql': 'SELECT "thing_SoAndSoModel"."id", '
         # …
         'WHERE "thing_SoAndSoModel"."related_object_id" = 0 LIMIT 21',
  'time': '0.001’}]
```
Which means that `rest_hooks` is only returning hooks where `user_id=1`.

That's my theory, anyway. If I have a moment later, I'll write a test to confirm.

**Comment by avelis** [comment]
@briankung If you find the time to make a pull request I can happily take a look and merge it in.

**Comment by briankung** [comment]
Hey, sorry, I couldn't devote enough time to figure out how to write a simple test for it. The function is difficult to test. I could submit a PR without a test, but that makes me uncomfortable.

**Comment by briankung** [comment]
Would it be correct to set user_override to False `if len(maybe_action) == 2`?

**Comment by avelis** [comment]
@briankung I believe you assessment is correct. If we changed `user_override=False` it should operate as expected. I still would try to get a unit test on it though.

I merged this enhancement in via a pull-request from another developer.

In an effort to be thorough I am CC'ing @GVRV to make sure such a change wouldn't create unintended consequences.

I doubtful that this is a consequence of using different Django versions.

I will still be looking into a way to confirm expected behavior via a unit test.

**Comment by imsickofmaps** [comment]
I can confirm this is a bug having spent two hours trying to confirm the same thing (without looking at the open issues! Idiot!). https://github.com/zapier/django-rest-hooks/issues/21#issuecomment-165599441 is the correct fix. The unit tests on this don't provide enough existing coverage to making adding a test just for this a short task! Are you still set on not patching without one?

**Comment by avelis** [comment]
@imsickofmaps I have pushed change in b3387f6fb5278796ef16be0b3c809ef835b0a64a Pull from master and let me know if that addresses this issue. I can then move to make a release.

**Comment by imsickofmaps** [comment]
👍  works for me... thanks!

**Comment by imsickofmaps** [comment]
@avelis can we have a release now?

**Comment by RadoRado** [comment]
Agreed - the changes work.

**Comment by PavlinGergov** [comment]
I can confirm the change fixes the problem.

**Comment by avelis** [comment]
Fixed and released v1.2.2. Release has also been pushed to PyPi.

See notes here:
https://github.com/zapier/django-rest-hooks/releases/tag/1.2.2

**Comment by imsickofmaps** [comment]
@avelis --> 
![you](https://cloud.githubusercontent.com/assets/1228977/15267377/eab6a3ee-19c0-11e6-9033-f5aadb078e2a.gif)
